cms/djangoapps/contentstore/views/item.py

```diff
@@ -1180,6 +1183,25 @@ def create_xblock_info(xblock, data=None, metadata=None, include_ancestor_info=F
     return xblock_info
 
 
+def get_group_xblock_name(user_partitions, xblock_display_name):
+    """
+    Get the group name if matching group xblock is found.
+
+    Arguments:
+        user_partitions (Dict): Locator of source item.
+        xblock_display_name (String): Display name of group xblock.
+
+    Returns:
+        group name (String): Group name of the matching group.
+    """
+    return [
+        group['name']
+        for user_partition in user_partitions
+        for group in user_partition['groups']
+        if str(group['id']) in xblock_display_name
+    ]
```

Comment:
Will this execute for all xblocks and don't we need to get display_name for given xblock_display_name, if yes then i think we should return the first match as it instead of executing nesting loop without any reason.